### PR TITLE
spiral-matrix: add to track

### PR DESCRIPTION
**Pull request by stkent**
[problem specification directory](https://github.com/exercism/problem-specifications/tree/57aa5f082262096d2bc5c983388ed5a00dc73822/exercises/spiral-matrix)

---

Reviewer Resources:

[Track Policies](https://github.com/exercism/xjava/blob/master/POLICIES.md#event-checklist)

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -285,6 +285,13 @@
       ]
     },
     {
+      "slug": "spiral-matrix",
+      "difficulty": 6,
```
Comment (stkent) [inline config.json]: Best guess.

File: exercises/spiral-matrix/src/main/java/SpiralMatrixBuilder.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class SpiralMatrixBuilder {
+
+
+
+}
```
Comment (stkent) [inline exercises/spiral-matrix/src/main/java/SpiralMatrixBuilder.java]: Check: is this needed per policy?

File: exercises/spiral-matrix/src/example/java/SpiralMatrixBuilder.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
+class SpiralMatrixBuilder {
+
+    int[][] ofSize(int size) {
+        if (size == 0) return new int[][]{};
```
Comment (FridaTveit) [inline exercises/spiral-matrix/src/example/java/SpiralMatrixBuilder.java]: I would add curly brackets and put the `if` body on a new line, just to be safe :)

File: exercises/spiral-matrix/src/main/java/SpiralMatrixBuilder.java (hunk 2 above)
Comment (FridaTveit) [inline exercises/spiral-matrix/src/main/java/SpiralMatrixBuilder.java]: No, according to the policy after the first 20 exercises no starter implementation needs to be provided. But quite a few exercises after the first 20 have an empty starter implementation like this. Do you think that's okay or should we make them more consistent?

File: exercises/spiral-matrix/src/example/java/SpiralMatrixBuilder.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
+class SpiralMatrixBuilder {
+
+    int[][] ofSize(int size) {
```
Comment (FridaTveit) [inline exercises/spiral-matrix/src/example/java/SpiralMatrixBuilder.java]: Maybe `buildMatrixOfSize()` would be a more descriptive name?

File: exercises/spiral-matrix/src/main/java/SpiralMatrixBuilder.java (hunk 2 above)
Comment (stkent) [inline exercises/spiral-matrix/src/main/java/SpiralMatrixBuilder.java]: I think there's an open issue for cleanup; let's decide there whether we want no implementation or empty class(es)! https://github.com/exercism/java/issues/554

File: exercises/spiral-matrix/src/example/java/SpiralMatrixBuilder.java (hunk 3 above)
Comment (stkent) [inline exercises/spiral-matrix/src/example/java/SpiralMatrixBuilder.java]: Kotlin habits sneaking in ;)

**Pull request merged**